### PR TITLE
Switch to GitHub actions for E2E tests

### PR DESCRIPTION
**Pull request by stevesoltys**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      - '**/*.xml'
+      - '.github/workflows/test.yml'
+
+jobs:
+  instrumentation_tests:
+    runs-on: macos-11
+    strategy:
+      fail-fast: false
+      matrix:
+        android_target: [33, 34]
```
Comment (chirayudesai) [inline .github/workflows/test.yml:16]: I like the idea of testing for 13 in the 14 branch, to make sure any changes we make here don't break stuff / are backwards compatible. As much as possible, at least. We'll try.

File: app/build.gradle [hunk 2]
```diff
@@ -189,7 +199,7 @@ tasks.register('provisionEmulator', Exec) {
     doFirst {
         commandLine "${project.projectDir}/development/scripts/provision_emulator.sh",
                 "seedvault",
-                "system-images;android-33;google_apis;x86_64"
+                "system-images;android-34;google_apis;x86_64"
```
Comment (chirayudesai) [inline app/build.gradle:202]: But I'd split this into a separate commit so we can easily backport the rest to 13

**Review by chirayudesai — DISMISSED** [review]
Minor nits, but I like, will be really nice to have it all be integrated.

**Comment by chirayudesai** [comment]
We could just merge this to `android14` directly. It'd fail right now, but then #576 should pass.

File: app/build.gradle (hunk 2 above)
Comment (stevesoltys) [reply app/build.gradle:202]: Done.

**Review by chirayudesai — APPROVED** [review]
